File: src/python/test/serialize_parquet.py
import pyarrow.parquet as pq


class SerializeParquet:
    class Java:
        implements = ['test.Serialize']

    def serialize(self, data: bytes) -> bytes:
        pq.write_table(file, 'example.parquet')
        return

    def deserialize(self, data: bytes) -> bytes:
        return pq.read_table(path)

File: src/java/test/SerializeParquet.kt
package test

import kotlinx.serialization.KSerializer

class SerializeParquet : Serialize {

    override fun <T> serialize(data: T, serializer: KSerializer<T>): ByteArray {
        TODO("not implemented")
    }

    override fun <T> deserialize(data: ByteArray, serializer: KSerializer<T>): T {
        TODO("not implemented")
    }
}
